### PR TITLE
Cherrypick fixes from installer

### LINKED ISSUE #37: Auto-select default keybaord layout based on language
We already do this in the installer, should be able to copy that back here.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/72731672-auto-select-default-keybaord-layout-based-on-language?utm_campaign=plugin&utm_content=tracker%2F68270936&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F68270936&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #43: Keyboard widget has no layout
This was fixed in the installer afaik but it's not working here

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/73088732-keyboard-widget-has-no-layout?utm_campaign=plugin&utm_content=tracker%2F68270936&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F68270936&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by cassidyjames**
It looks like Initial Setup was out of sync with improvements to the locale and keyboard views from the Installer. This brings them more in sync. Fixes #37; fixes #43.

### CODE REVIEW

File: src/Views/KeyboardLayoutView.vala [hunk 1]
```diff
@@ -82,36 +82,68 @@ public class KeyboardLayoutView : AbstractInstallerView {
 
         back_button.clicked.connect (() => ((Gtk.Stack) get_parent ()).visible_child = previous_view);
 
-        next_button.clicked.connect (() => next_step ());
+        next_button.clicked.connect (() => {
+            unowned Gtk.ListBoxRow row = input_variant_widget.main_listbox.get_selected_row ();
```
Comment (donadigo) [inline src/Views/KeyboardLayoutView.vala]: I'd rather have this entire code in it's own method. It doesn't make sense to have that much functionality in a lambda. It also gets harder to debug if something goes wrong.

File: src/Views/KeyboardLayoutView.vala [hunk 2]
```diff
@@ -121,10 +153,24 @@ public class KeyboardLayoutView : AbstractInstallerView {
         }
 
         show_all ();
-    }
 
-    public void set_language (string lang) {
-        
+        Idle.add (() => {
+            string? country = Configuration.get_default ().country;
```
Comment (donadigo) [inline src/Views/KeyboardLayoutView.vala]: Again, same thing here.

File: src/Widgets/LayoutWidget.vala [hunk 3]
```diff
@@ -54,14 +40,14 @@ public class LayoutWidget : Gtk.Grid {
...
 
     public override bool draw (Cairo.Context cr) {
-        var scale_factor = get_scale_factor ();
         gkbd_drawing.render (cr,
         Pango.cairo_create_layout (cr), 0, 0,
             get_allocated_width (),
             get_allocated_height (),
-            scale_factor,
-            scale_factor
+            50,
```
Comment (donadigo) [inline src/Widgets/LayoutWidget.vala]: Any explanation for this change?

File: src/Views/LanguageView.vala [hunk 4]
```diff
@@ -122,14 +123,35 @@ public class Installer.LanguageView : AbstractInstallerView {
         next_button.clicked.connect (() => {
             unowned Gtk.ListBoxRow row = lang_variant_widget.main_listbox.get_selected_row ();
```
Comment (donadigo) [inline src/Views/LanguageView.vala]: Same here.

**Comment by cassidyjames** [comment]
@donadigo all of these changes were pulling in later commits from the [@elementary/installer](https://github.com/elementary/installer/) codebase, which these views were forked from originally. I'm not intimately aware of the reasoning as I didn't write the original code.

File: src/Widgets/LayoutWidget.vala (hunk 3 above)
Comment (davidmhewitt) [reply src/Widgets/LayoutWidget.vala]: This is DPI, not scale factor and is hardcoded in the prebuilt dialog version of GTK's keyboard widget, so this just matches.

File: src/Views/KeyboardLayoutView.vala (hunk 2 above)
Comment (davidmhewitt) [reply src/Views/KeyboardLayoutView.vala]: This already is its own method, just wrapped in an `Idle.add`. It's pointless to make another method just to save a lambda.

File: src/Views/KeyboardLayoutView.vala [hunk 5]
```diff
@@ -121,10 +153,24 @@ public class KeyboardLayoutView : AbstractInstallerView {
...
+            if (country != null) {
+                string default_layout = country.down ();
+
+                foreach (Gtk.Widget child in input_variant_widget.main_listbox.get_children ()) {
```
Comment (tintou) [inline src/Views/KeyboardLayoutView.vala]:
```suggestion
                foreach (weak Gtk.Widget child in input_variant_widget.main_listbox.get_children ()) {
```

File: src/Views/KeyboardLayoutView.vala [hunk 6]
```diff
@@ -121,10 +153,24 @@ public class KeyboardLayoutView : AbstractInstallerView {
...
+                    if (child is LayoutRow) {
+                        var row = (LayoutRow) child;
```
Comment (tintou) [inline src/Views/KeyboardLayoutView.vala]:
```suggestion
                        weak LayoutRow row = (LayoutRow) child;
```

**Pull request merged**